test/cpp/api/jit.cpp

```diff
@@ -85,7 +85,16 @@ TEST(TorchScriptTest, TestNestedIValueModuleArgMatching) {
                   "position 0, but instead got value of type Tensor[][]") == 0);
 
   };
+}
 
 
-
+TEST(TorchScriptTest, TestDictArgMatching) {
+  auto module = torch::jit::compile(R"JIT(
+      def dict_op(a: Dict[str, Tensor], b: str):
+        return a[b]
+    )JIT");
+  c10::ivalue::DictUnorderedMap<torch::jit::IValue, torch::jit::IValue> dict;
```

Comment:
Can we add conversion for unordered_map<std::string, Tensor> ?

Reply:
I'll leave that for a follow up since it involves a bunch changes to `IValue`